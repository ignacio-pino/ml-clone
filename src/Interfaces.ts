export interface IProductListData {
  id: string;
  title: string;
  price: string;
  original_price?: string;
  currency_id: "UYU" | "USD";
  thumbnail: string;
  shipping: {
    free_shipping: boolean;
  };
}

export interface ISortDataList {
  sort: ISortData;
  available_sorts: ISortData[];
}

export interface ISortData {
  id: string;
  name: string;
}
export interface IProductCard {
  id: string;
  title: string;
  price: string;
  original_price?: string;
  currency_id: "UYU" | "USD";
  thumbnail: string;

  free_shipping: boolean;
}

export interface IFilterDataList {
  filters: IFilterList[];
  availableFilters: IFilterList[];
}
export interface IFilterList {
  id: string;
  name: string;
  type: string;
  values: IFilterListValues[];
}

export interface IFilterListValues {
  id: string;
  name: string;
  results?: number;
  path_from_root?: IPathFromRoot[];
}

export interface ICategoryData {
  query?: string;
  categoryInformation?: ICategoryInformation;
  totalResults: number;
  childrenCategories?: IPathFromRoot[];
}

export interface ICategoryInformation {
  id: string;
  name: string;
  path_from_root?: IPathFromRoot[];
}

export interface IPathFromRoot {
  id: string;
  name: string;
}

export interface IChildrenCategoryData {
  id: string;
  name: string;
  total_items_in_this_category: number;
}

export interface IHeaderCategoryData {
  categoryId: string;
  categoryName: string;
}
