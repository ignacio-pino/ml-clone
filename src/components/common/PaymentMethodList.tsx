import { InfoIcon } from "@chakra-ui/icons";
import { VStack, Image, Text, HStack } from "@chakra-ui/react";
import { IPaymentMethods } from "../../Interfaces";

export default function PaymentMethodList({
  paymentMethods,
}: {
  paymentMethods: IPaymentMethods[];
}) {
  return (
    <VStack
      alignItems={"flex-start"}
      border={"1px"}
      borderColor={"gray.500"}
      borderRadius={"10px"}
      p={"1vw"}
      gap={3}
    >
      <Text size={"l"}>Medios de pago</Text>
      <HStack backgroundColor={"purple.400"} borderRadius={"5px"} p={"0.5rem"}>
        <InfoIcon />
        <Text>¡Paga en hasta 10 cuotas sin interés! </Text>
      </HStack>
      <Text as="b">Tarjetas de Crédito</Text>
      <Text color={"gray.500"} fontSize={"sm"} mt={-2}>
        ¡Cuotas sin interés con bancos seleccionados!
      </Text>
      <HStack gap={3} flexWrap={"wrap"}>
        {paymentMethods.map(({ payment_type_id, id, thumbnail }) =>
          payment_type_id === "credit_card" ? (
            <Image
              key={id}
              src={thumbnail}
              mixBlendMode={"multiply"}
              maxW={"60px"}
              maxH={"60px"}
              alt={thumbnail}
            />
          ) : null
        )}
      </HStack>
      <Text as="b">Tarjetas de Débito</Text>
      <HStack gap={3}>
        {paymentMethods.map(({ payment_type_id, id, thumbnail }) =>
          payment_type_id === "debit_card" ? (
            <Image
              key={id}
              src={thumbnail}
              mixBlendMode={"multiply"}
              maxW={"60px"}
              maxH={"60px"}
              alt={thumbnail}
            />
          ) : null
        )}
      </HStack>
      <Text as="b">Efectivo</Text>
      <HStack gap={3}>
        {paymentMethods.map(({ payment_type_id, id, thumbnail }) =>
          payment_type_id === "ticket" ? (
            <Image
              key={id}
              src={thumbnail}
              mixBlendMode={"multiply"}
              maxW={"60px"}
              maxH={"60px"}
              alt={thumbnail}
            />
          ) : null
        )}
      </HStack>
    </VStack>
  );
}
