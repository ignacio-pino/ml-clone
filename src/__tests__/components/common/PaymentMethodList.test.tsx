import { render, screen } from "@testing-library/react";
import PaymentMethodList from "../../../components/common/PaymentMethodList";
import { IPaymentMethods } from "../../../Interfaces";

describe("PaymentMethodList", () => {
  const samplePaymentMethods: IPaymentMethods[] = [
    {
      payment_type_id: "credit_card",
      id: "1",
      thumbnail: "image1.jpg",
    },
    {
      payment_type_id: "debit_card",
      id: "2",
      thumbnail: "image2.jpg",
    },
    {
      payment_type_id: "ticket",
      id: "3",
      thumbnail: "image3.jpg",
    },
  ];

  it("should match snapshot", () => {
    const container = render(
      <PaymentMethodList paymentMethods={samplePaymentMethods} />
    ).container;

    expect(container).toMatchSnapshot();
  });

  it("renders credit card payment methods correctly", () => {
    render(<PaymentMethodList paymentMethods={samplePaymentMethods} />);

    const creditCardImage = screen.getByAltText(
      "image1.jpg"
    ) as HTMLImageElement;
    expect(creditCardImage.src.endsWith("image1.jpg")).toBe(true);
  });

  it("renders debit card payment methods correctly", () => {
    render(<PaymentMethodList paymentMethods={samplePaymentMethods} />);

    const debitCardImage = screen.getByAltText(
      "image2.jpg"
    ) as HTMLImageElement;
    expect(debitCardImage.src.endsWith("image2.jpg")).toBe(true);
  });

  it("renders cash payment methods correctly", () => {
    render(<PaymentMethodList paymentMethods={samplePaymentMethods} />);

    const ticketImage = screen.getByAltText("image3.jpg") as HTMLImageElement;
    expect(ticketImage.src.endsWith("image3.jpg")).toBe(true);
  });

  it("renders the payment method information", () => {
    render(<PaymentMethodList paymentMethods={samplePaymentMethods} />);

    const creditCardText = screen.getByText("Tarjetas de Crédito");
    expect(creditCardText).toBeInTheDocument();

    const debitCardText = screen.getByText("Tarjetas de Débito");
    expect(debitCardText).toBeInTheDocument();

    const ticketText = screen.getByText("Efectivo");
    expect(ticketText).toBeInTheDocument();
  });
});
