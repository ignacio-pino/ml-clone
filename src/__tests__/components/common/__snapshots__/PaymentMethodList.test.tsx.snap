// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PaymentMethodList > should match snapshot 1`] = `
<div>
  <div
    class="chakra-stack css-pbh4bz"
  >
    <p
      class="chakra-text css-0"
    >
      Medios de pago
    </p>
    <div
      class="chakra-stack css-szbcth"
    >
      <svg
        class="chakra-icon css-onkibi"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12,0A12,12,0,1,0,24,12,12.013,12.013,0,0,0,12,0Zm.25,5a1.5,1.5,0,1,1-1.5,1.5A1.5,1.5,0,0,1,12.25,5ZM14.5,18.5h-4a1,1,0,0,1,0-2h.75a.25.25,0,0,0,.25-.25v-4.5a.25.25,0,0,0-.25-.25H10.5a1,1,0,0,1,0-2h1a2,2,0,0,1,2,2v4.75a.25.25,0,0,0,.25.25h.75a1,1,0,1,1,0,2Z"
          fill="currentColor"
        />
      </svg>
      <p
        class="chakra-text css-0"
      >
        ¡Paga en hasta 10 cuotas sin interés! 
      </p>
    </div>
    <b
      class="chakra-text css-0"
    >
      Tarjetas de Crédito
    </b>
    <p
      class="chakra-text css-2m19dl"
    >
      ¡Cuotas sin interés con bancos seleccionados!
    </p>
    <div
      class="chakra-stack css-1xnim7v"
    >
      <img
        alt="image1.jpg"
        class="chakra-image css-sfrnhg"
        src="image1.jpg"
      />
    </div>
    <b
      class="chakra-text css-0"
    >
      Tarjetas de Débito
    </b>
    <div
      class="chakra-stack css-12cfr4k"
    >
      <img
        alt="image2.jpg"
        class="chakra-image css-sfrnhg"
        src="image2.jpg"
      />
    </div>
    <b
      class="chakra-text css-0"
    >
      Efectivo
    </b>
    <div
      class="chakra-stack css-12cfr4k"
    >
      <img
        alt="image3.jpg"
        class="chakra-image css-sfrnhg"
        src="image3.jpg"
      />
    </div>
  </div>
</div>
`;
